gst/droidcamsrc/gstdroidcamsrc.c

```diff
@@ -482,29 +479,15 @@ gst_droidcamsrc_get_hw (GstDroidCamSrc * src)
     return FALSE;
   }
 
-  if (num > MAX_CAMERAS) {
-    GST_WARNING_OBJECT (src, "cannot support %d cameras", num);
-  }
-
-  src->info[0].num = src->info[1].num = -1;
+  src->info = g_malloc0 (num * sizeof (GstDroidCamSrcCamInfo));
 
-  back_found =
-      gst_droidcamsrc_fill_info (src, &src->info[0],
-      DROID_MEDIA_CAMERA_FACING_BACK);
-  if (!back_found) {
-    GST_WARNING_OBJECT (src, "cannot find back camera");
-  }
-
-  front_found =
-      gst_droidcamsrc_fill_info (src, &src->info[1],
-      DROID_MEDIA_CAMERA_FACING_FRONT);
-  if (!front_found) {
-    GST_WARNING_OBJECT (src, "cannot find front camera");
-  }
+  for (x = 0; x < num; x++) {
+    gboolean found;
 
-  if (!front_found && !back_found) {
-    GST_ERROR_OBJECT (src, "no cameras found");
-    return FALSE;
+    found = gst_droidcamsrc_fill_info (src, &src->info[x], x);
+    if (!found) {
+      GST_WARNING_OBJECT (src, "cannot find camera %d", x);
```

Comment:
If there's an issue with an item at an index it should probably be removed from the set of cameras.

Reply:
Yeah, we should fix this later. There is slightly small change that this will cause problem. I'll merge this now as qtmultimedia got already merged.